js/ui/overview.js

```diff
@@ -525,8 +525,22 @@ const Overview = new Lang.Class({
         this.viewSelector.focusSearch();
     },
 
-    _onStartupCompleted: function() {
-        this.showApps();
+    _isInitialSetupRunning: function() {
```

Comment:
I wonder if we couldn't instead more directly check the session mode to determine this information.

Reply:
Hmm... that's a good point, but your suggestion would work fine for the "true FBE", the one that happens only once on the very first boot, but would not work for FBEs triggered for new users, since gnome shell wouldn't be running in initial-setup mode for them.

That said, we are actively disabling FBE for existing installations (see T19802), so perhaps that's a fair assumption to make and we can rely on the session mode. Or, even better, perhaps we can drop this initial-setup related bits altogether since, now I'm thinking twice, I'm having doubts of the impact of changing the viewSelector to point to the APPs page if the shell is running in a different mode anyway... I'll test this theory and push a new patch

Reply:
Indeed, no need to do this at all, as it's only useful for `--existing-user` runs of FBE, which we disable. Sorry for the confusion, and thanks for pointing it out!

Reply:
Good point about gnome-initial-setup running in existing user mode, even though we don't use it... Feel free to ignore my suggestion if this seems more practical.